PyQT5_fbs/src/main/python/my_widgets.py

```diff
@@ -422,6 +423,14 @@ def calculate_and_plot_residuals(self, sens_str1, sens_str2, mode):
             data_hr = filt.hr_process(data_obj, datetime(year, month, 1, 0, 0, 0),
                                         datetime(year_end, month_end + 1, 1, 0, 0, 0))
 
+            # Subtract the mean from the sensor data for comparison with new tide prediction.
+            if sens_str1 == 'PRD':
+                sensor_data_flat = data_hr[sens_str2.lower()]["sealevel"].flatten()
+                sensor_data_flat_filt = [x for x in sensor_data_flat if not math.isnan(x)]
+                sensor_data_mean = sum(sensor_data_flat_filt) / len(sensor_data_flat_filt)
```

Comment:
Nit: Instead of filtering and finding mean you could just use np.nanmean(sensor_data_flat) and save a line of code and some CPU cycles :)

Reply:
Approved and implemented : )